modules/core/shared/src/main/scala/lucuma/core/util/TimeSpan.scala

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.util
+
+import cats.Order
+import cats.Order.catsKernelOrderingForOrder
+import cats.syntax.option.*
+import cats.syntax.order.*
+import eu.timepit.refined.types.numeric.NonNegLong
+import lucuma.core.optics.Format
+import monocle.Iso
+import monocle.Prism
+
+import java.math.RoundingMode.HALF_UP
+import java.time.Duration
+import java.time.temporal.ChronoUnit.MICROS
+import scala.util.Try
+
+/**
+ * TimeSpan is a time span, similar to a `FiniteDuration` in that it is a
+ * single `Long` value representing an amount of time, but in this case fixed
+ * to positive microseconds. This corresponds with the ODB GraphQL schema
+ * duration, which allows setting and displaying times in Long microseconds
+ * and fractional milliseconds, seconds, minutes, and hours.
+ */
+opaque type TimeSpan = NonNegLong
+
+object TimeSpan {
+
+  val Min: TimeSpan =
+    NonNegLong.MinValue
+
+  val Max: TimeSpan =
+    NonNegLong.MaxValue
+
+  /**
+   * Constructs a TimeSpan from the given number of microseconds, if it is
+   * non-negative.
+   */
+  def fromMicroseconds(µs: Long): Option[TimeSpan] =
+    NonNegLong.from(µs).toOption
+
+  def unsafeFromMicroseconds(µs: Long): TimeSpan =
+    fromMicroseconds(µs).getOrElse(sys.error(s"The µs value ($µs) must be non-negative."))
+
+  def fromNonNegMicroseconds(µs: NonNegLong): TimeSpan =
```

Comment:
I guess this could be `apply` or have another `apply` method?

Reply:
:+1: added `apply`.